### DISCUSSION TITLE
Echo Auth propagate token to request.Context

### DISCUSSION THREAD

**Discussion by gianarb** [oapi-codegen/oapi-codegen#1042] [Q&A]
Hello, thank you for the hard work.

I am following this example https://github.com/deepmap/oapi-codegen/blob/29ebe3262399dbbf9d9ce612757e50cd2ec2862f/examples/authenticated-api/echo/server/jwt_authenticator.go#L82 but I would like to propagate the token as part of the echoContext.Request().Context() as well because I propagate the std context in the underline layers of my app and I don't want to set the token manually in every http handler.

I tried with something like this:

```
eCtx.SetRequest(eCtx.Request().WithContext(context.WithValue(eCtx.Request().Context(), "AuthInfo", info)))
```

But I get an error when I use `ectx.Bind` in the actual api handler `{"message":"code=400, message=http: invalid Read on closed Body, internal=http: invalid Read on closed Body"}   `

I am wondering what is the proper way to go. Thanks

**Comment by ianmuhia** [comment]
facing the same issue did you manage to solve this @gianarb
